File: xlspaceship-battle/src/main/webapp/resources/js/controllers/BookController.js
'use strict';

var bookApp = angular.module('bookApp', []);
bookApp.controller('bookController', function($scope, $http) {
	
	$scope.getBooks = function() {	
		$http({
			method: 'GET', 
			url: "book"
		}).then(function(response) {
			console.log("BOOK_LIST:" + response.data);
			$scope.books = response.data;
		});
	};
	
	$scope.getBook = function(bookId) {	
		$http({
			method: 'GET', 
			url: "book/" + bookId
		}).then(function(response) {
			console.log(response);
			$scope.books = response.data;
		});
	};
	
	$scope.addBook = function() {    	
		$http({
			method: 'POST', 
			url: "book/add", 
			data: {
	    		title : $scope.bookForm.title,
	    		author : $scope.bookForm.author,
	    		category : $scope.bookForm.category
	        }, 
		}).then(function(response) {
			console.log("BOOK_ADDED: " + response.data);
			$scope.getBooks();
			$scope.resetBookForm();
		});
    };
    
    $scope.updateBook = function() {    	
		$http({
			method: 'PUT', 
			url: "book/update", 
			data: {
				bookId : $scope.bookForm.bookId,
				title : $scope.bookForm.title,
	    		author : $scope.bookForm.author,
	    		category : $scope.bookForm.category
	        }, 
		}).then(function(response) {
			console.log("BOOK_UPDATED: " + response.data);
			$scope.getBooks();
			$scope.resetBookForm();
		});
    };
    
    $scope.deleteBook = function(bookId) {	
		$http({
			method: 'DELETE', 
			url: "book/" + bookId
		}).then(function(response) {
			console.log("BOOK_DELETED: " + response.data);
			$scope.getBooks();
			$scope.resetBookForm();
		});
	};
	
	$scope.editBook = function(bookId) {	
		$http({
			method: 'GET', 
			url: "book/" + bookId
		}).then(function(response) {
			console.log("BOOK_EDIT: " + response.data);
			$scope.bookForm = {
				bookId : response.data.bookId,
				title : response.data.title,
				author : response.data.author,
				category : response.data.category
			};
		});
	};
    
	$scope.saveBook = function() {
		console.log("BOOK_SAVE - bookID: " + $scope.bookForm.bookId);
		var bookID = $scope.bookForm.bookId;
		if (bookID === null) {
			$scope.addBook();
		} else {
			$scope.updateBook();
		}
	};
	
    $scope.resetBookForm = function() {
    	
    	$scope.bookForm = {
			bookId : null,
			title : null,
			author : null,
			category : null
		};
    };
    
    $scope.resetBookForm();
    $scope.getBooks();
	
});

File: xlspaceship-battle/src/main/java/com/xlspaceship/battle/controller/BookController.java
package com.xlspaceship.battle.controller;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

import com.xlspaceship.battle.form.BookForm;
import com.xlspaceship.battle.model.Book;
import com.xlspaceship.battle.service.BookService;

@RestController
@RequestMapping("/book")
public class BookController {
	
	@Autowired
	private BookService bookService; 
	
	@RequestMapping(method = RequestMethod.GET)
	public ResponseEntity<List<Book>> getBooks() {
		List<Book> books = bookService.getBooks();
		return ResponseEntity.ok().body(books);
	}
	
	@RequestMapping(value = "/{bookId}", method = RequestMethod.GET)
	public ResponseEntity<Book> getBook(@PathVariable("bookId") String bookId) {
		Book book = bookService.getBook(Long.valueOf(bookId));
		return ResponseEntity.ok().body(book);
	}
	
	@RequestMapping(value = "/add", method = RequestMethod.POST)
	public void addBook(@RequestBody BookForm bookForm) {
		Book book = new Book();
		book.setTitle(bookForm.getTitle());
		book.setAuthor(bookForm.getAuthor());
		book.setCategory(Integer.valueOf(bookForm.getCategory()));
		bookService.addBook(book);
	}

	@RequestMapping(value = "/update", method = RequestMethod.PUT)
	public void updateBook(@RequestBody BookForm bookForm) {
		Book book = new Book();
		book.setBookId(Long.valueOf(bookForm.getBookId()));
		book.setTitle(bookForm.getTitle());
		book.setAuthor(bookForm.getAuthor());
		book.setCategory(Integer.valueOf(bookForm.getCategory()));
		bookService.updateBook(book);
	}
	
	@RequestMapping(value = "/{bookId}", method = RequestMethod.DELETE)
	public void deleteBook(@PathVariable("bookId") String bookId) {
		bookService.deleteBook(Long.valueOf(bookId));
	}
	
}
